DataAccessLayer/Repositories/Implementation/CategoryRepository.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using DataAccessLayer.DataBaseModels;
+using DataAccessLayer.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccessLayer.Repositories.Implementation
+{
+    public class CategoryRepository : ICrudRepository<Category>
+    {
+
+        public List<Category> GetAll()
+        {
+            using LanguageSkillsDBContext db = new LanguageSkillsDBContext();
+            return db.Categories.ToList();
+        }
+            
+
+        public Category Get(int id)
+        {
+            try
+            {
+                using LanguageSkillsDBContext db = new LanguageSkillsDBContext();
```

Comment:
Wrong usage of "using".
Please how it's used here: https://metanit.com/sharp/entityframework/1.2.php
Or here: https://stackoverflow.com/questions/15666824/entity-framework-and-calling-context-dispose

Reply:
Also relevant to all other cases where "using" DB context